packages/flutter/lib/src/animation/curves.dart

```diff
@@ -267,6 +273,526 @@ class Cubic extends Curve {
+  /// restrictions placed on control points.
+  ///
+  /// The optional [tension] argument controls how tightly the curve approaches
+  /// the given `points`. It must be in the range 0.0 to 1.0, inclusive. It
+  /// defaults to 0.0, which provides the smoothest curve. A value of 1.0
+  /// is equivalent to a linear interpolation between points.
+  ///
+  /// All of the arguments must not be null.
+  ///
+  /// See also:
+  ///
+  ///  * This [paper on using Catmull-Rom splines](http://faculty.cs.tamu.edu/schaefer/research/cr_cad.pdf).
+  CatmullRomCurve(this.controlPoints, {this.tension = 0.0})
+      : assert(tension != null),
+        assert(() {
+          return validateControlPoints(
+            controlPoints,
+            tension: tension,
+            reasons: _assertReasons..clear(),
+          );
+        }(), 'control points $controlPoints could not be validated:\n  ${_assertReasons.join('\n  ')}'),
+        // Pre-compute samples so that we don't have to evaluate the spline's inverse
+        // all the time in transformInternal.
+        _precomputedSamples = CatmullRomSpline(
+          // Force the first and last control points for the spline to be (0, 0)
+          // and (1, 1), respectively.
+          <Offset>[Offset.zero, ...controlPoints, const Offset(1.0, 1.0)],
+          tension: tension,
+        ).generateSamples(start: 0.0, end: 1.0, tolerance: 1e-12);
+
+  /// A static accumulator for assertion failures. Not used in release mode.
+  static final List<String> _assertReasons = <String>[];
+
+  // The precomputed approximation curve, so that evaluation of the curve is
+  // efficient.
+  final List<Curve2DSample> _precomputedSamples;
+
+  /// The control points used to create this curve.
+  ///
+  /// The `dx` value of each [Offset] in [controlPoints] represents the
+  /// animation value at which the curve should pass through the `dy` value of
+  /// the same control point.
+  ///
+  /// The [controlPoints] list must meet the following criteria:
+  ///
+  ///  * The `points` and `tension` arguments must not be null.
+  ///  * The list of `points` must contain at least two points.
+  ///  * The X value of each point must be greater than 0.0 and less then 1.0.
+  ///  * The X values of each point must be greater than the
+  ///    previous point's X value (i.e. monotonically increasing). The Y values
+  ///    are not constrained.
+  ///  * The resulting spline must be single-valued in X. That is, for each X
+  ///    value, there must be exactly one Y value. This means that the control
+  ///    points must not generated a spline that loops or overlaps itself.
+  ///
+  /// The static function [validateControlPoints] can be used to check that
+  /// these conditions are met, and will return true if they are. In debug mode,
+  /// it will also optionally return a list of reasons in text form. In debug
+  /// mode, this constructor will assert that these conditions are met and print
```

Comment:
"this constructor" -> constructor of CatmullRomCurve?

Reply:
Yes. Fixed.